### PR TITLE
Manage anchor proposals

### LINKED ISSUE #39: Manage anchor proposals
If an anchor has been proposed, but is missing signatures, it should be shown in the org view.

* On `/orgs/:address`, there should be a section that shows proposed anchors
* Only org members should see the proposed anchors
* The page should allow members to add their signature/confirmation to the anchor proposal
* The number of required signatures should be shown
* The members who signed the proposal should be shown

### Notes

* `SafeServiceClient.getPendingTransactions` to get proposals
* `SafeServiceClient.decodeData` to find out if the tx is an anchor proposal
* `SafeServiceClient.confirmTransaction` to sign a proposal
* To execute the proposal, we can use the SafeCoreSDK

### PR DESCRIPTION
**Pull request by sebastinez**
This PR closes #39 

It allows the signing and execution of anchor proposals for gnosis safe orgs.

**Some interesting points:**
- At the moment a pending project requires at least the required confirmations, once received the threshold, we don't accept more confirmations, but allow to execute the safe transaction.
- The owners that signed the proposal, show up as blockies, we could eventually show the profile avatar or show something entirely different, IMO the blockies maintain the design of the rest of the page.

### CODE REVIEW

File: src/base/orgs/Org.ts [hunk 1]
```diff
@@ -174,6 +180,40 @@ export class Org {
...
+    const orgAddr = ethers.utils.getAddress(this.address);
+    const response = await config.safe.client.getPendingTransactions(
+      ethers.utils.getAddress(this.owner)
+    );
+    const projects: PendingProject[] = [];
+
+    for (const tx of response.results || []) {
+      if (tx.data && tx.to === orgAddr) {
+        const project = parseAnchorTx(tx.data, config);
+
```
Comment (cloudhead) [inline src/base/orgs/Org.ts:195]:
```suggestion
```

File: src/base/orgs/Org.ts [hunk 2]
```diff
@@ -336,3 +379,20 @@ export class Org {
     });
   }
 }
+
+export function parseAnchorTx(data: string, config: Config): Project | undefined {
```
Comment (cloudhead) [inline src/base/orgs/Org.ts:383]: Would prefer to return `null` here, as discussed, to be consistent.

File: src/base/orgs/View/Anchor.svelte [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  let state = State.Idle;
+  let action: null | Action = null;
+  const isSigned = project.confirmations.includes(
+    ethers.utils.getAddress(session.address)
+  );
+  const close = () => {
+    action = null;
+    state = State.Idle;
+  };
+  const pending = safe.threshold - project.confirmations.length;
```
Comment (cloudhead) [inline src/base/orgs/View/Anchor.svelte:40]: I would check the confirmations is `<` than threshold first. It's possible for there to be more confirmations than the threshold.

Comment (cloudhead) [reply src/base/orgs/View/Anchor.svelte:40]: Nevermind, you do a `<=` check lower.

File: src/base/orgs/View/Anchor.svelte [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    <span slot="actions">
+      {#if state == State.Confirm}
+        <button class="primary" on:click={() => confirmAnchor(project.safeTxHash)}>
+          Confirm
+        </button>
+        <button class="text" on:click={close}>
+          Cancel
+        </button>
+      {:else if state == State.Success}
+        <!-- TODO: See if there is a better way to refresh the state in the upper components. -->
```
Comment (cloudhead) [inline src/base/orgs/View/Anchor.svelte:166]: You could use `dispatch` to dispatch an event.

Comment (sebastinez) [reply src/base/orgs/View/Anchor.svelte:166]: Good idea, this would be better :+1:

**Review by cloudhead — APPROVED** [review]
Nice :+1:

**Pull request merged**